site-manage/site-manage-impl/impl/src/java/org/sakaiproject/sitemanage/impl/SiteManageServiceImpl.java

```diff
@@ -111,9 +112,21 @@ public boolean importToolsIntoSiteThread(final Site site, final List<String> exi
         Runnable siteImportTask = () -> {
             sessionManager.setCurrentSession(session);
             sessionManager.setCurrentToolSession(toolSession);
-            eventTrackingService.post(eventTrackingService.newEvent(SiteService.EVENT_SITE_IMPORT_START, site.getReference(), false));
-
-            try {
+            
+			String importSites ="";
+			for (Map.Entry<String, List<String>> entry : importTools.entrySet()) {
+				if(importSites.length() >= 255)
+					break;
+				for(String data : entry.getValue() ) {
+					if(!importSites.contains(data) && importSites.concat(data + ", ").length() < 255)
+						importSites = importSites.concat(data + ", ");
```

Comment:
You can use `StringUtils.join()` here, so you don't have to worry about removing the trailing ", " afterwards.

Reply:
@bjones86 I think String.join() removes code on the one hand, but adds another because it is not a simple list, is a map with lists so you have to do a join for any list of the map and concat them. Then it will still be necessary to verify if there is information in importSites or not  to add the "," in that concat. 

Also is necessary to verify the length of the String that does not exceed 255, using the String.join() the result of the join can be limited, but incomplete elements may remain.

Reply:
I'm not suggesting to remove the `< 255` checks; but simply to replace:

```
importSites.concat(data + ", ")
```

with:

```
StringUtils.joinWith(", ", importSites, data)
```

In this way, the checks to validate the string is not greater than 255 characters are still in place, but lines 127-129 become unnecessary.